Mathlib/CategoryTheory/LiftingProperties/Basic.lean

```diff
@@ -127,6 +127,17 @@ theorem iff_of_arrow_iso_right {A B X Y X' Y' : C} (i : A ⟶ B) {p : X ⟶ Y} {
   constructor <;> intro
   exacts [of_arrow_iso_right i e, of_arrow_iso_right i e.symm]
 
+variable {i p} (h : HasLiftingProperty i p)
+  {f : A ⟶ X} {g : B ⟶ Y} (sq : CommSq f i p g)
+
+/-- A choice of a diagonal morphism that is part of a `LiftStruct` when
+the square has a lift. -/
+noncomputable abbrev lift : B ⟶ X := sq.lift
+
+theorem fac_left : i ≫ h.lift sq = f := sq.fac_left
+
+theorem fac_right : h.lift sq ≫ p = g := sq.fac_right
+
```

Comment:
This should be removed. (I had forgotten we had `CommSq.lift`.) As `HasLiftingProperty` is a typeclass, we should use it as much as possible as a `[...]` parameter.